pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -116,16 +116,20 @@ func (f *ShardAwareDeduplicateFilter) Filter(ctx context.Context, metas map[ulid
 // So for example there is only one ULID(9) node, referenced from nodes 5, 6, 7, 8 (each of them also exists only once). See
 // blockWithSuccessors structure -- it uses maps to pointers to handle all this cross-referencing correctly.
 func (f *ShardAwareDeduplicateFilter) findDuplicates(ctx context.Context, input []*metadata.Meta) (map[ulid.ULID]struct{}, error) {
+	// We create a tree of blocks with successors (blockWithSuccessors) by
+	// 1) sorting the input blocks by number of sources, and
+	// 2) iterating through each input block, and adding it to the correct place in the tree of blocks with successors.
+
 	// Sort blocks with fewer sources first.
 	sort.Slice(input, func(i, j int) bool {
-		ilen := len(input[i].Compaction.Sources)
-		jlen := len(input[j].Compaction.Sources)
+		iLen := len(input[i].Compaction.Sources)
```

Comment:
nit: we try to avoid superficial changes like this if there's no lint for it because it makes `git blame` less useful.